deploy/helm/elastic-agent/templates/agent/_helpers.tpl

```diff
@@ -0,0 +1,412 @@
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "elasticagent.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "elasticagent.fullname" -}}
+{{- if .Values.fullnameOverride }}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default .Chart.Name .Values.nameOverride }}
+{{- if contains $name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "elasticagent.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Entrypoint for chart initialisation
+*/}}
+{{- define "elasticagent.init" -}}
+{{- if not (hasKey $.Values.agent "initialised") -}}
+{{/* init order matters */}}
+{{- include "elasticagent.init.engine" $ -}}
+{{- include "elasticagent.init.fleet" $ -}}
+{{- include "elasticagent.init.outputs" $ -}}
+{{- include "elasticagent.init.inputs" $ -}}
+{{- include "elasticagent.init.presets" $ -}}
+{{- $_ := set $.Values.agent "initialised" dict -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Check the agent.engine and fallback to "k8s"
+*/}}
+{{- define "elasticagent.init.engine" -}}
+{{- $ := . -}}
+{{- if empty (dig "engine" ("") $.Values.agent) -}}
+{{- $_ := set $.Values.agent "engine" "k8s" -}}
+{{- end -}}
+{{- include (printf "elasticagent.engine.%s.init" $.Values.agent.engine) $ -}}
+{{- end -}}
+
+{{/*
+Validate fleet configuration
+*/}}
+{{- define "elasticagent.init.fleet" -}}
+{{- $ := . -}}
+{{- if eq $.Values.agent.fleet.enabled true -}}
+{{- if empty $.Values.agent.fleet.url -}}
+{{- fail "url must be defined when fleet mode is enabled" -}}
+{{- end -}}
+{{- if empty $.Values.agent.fleet.token -}}
+{{- if empty $.Values.agent.fleet.kibanaHost -}}
+{{- fail "kibana host must be defined when fleet mode is enabled and a token is not supplied" -}}
+{{- end -}}
+{{- if empty $.Values.agent.fleet.kibanaUser -}}
+{{- fail "kibana username must be defined when fleet mode is enabled and a token is not supplied" -}}
+{{- end -}}
+{{- if empty $.Values.agent.fleet.kibanaPassword -}}
+{{- fail "kibana password must be defined when fleet mode is enabled and a token is not supplied" -}}
+{{- end -}}
+{{- end -}}
+{{- if empty $.Values.agent.fleet.preset -}}
+{{- fail "preset must be defined when fleet mode is enabled" -}}
+{{- end -}}
+{{- if not (hasKey $.Values.agent.presets $.Values.agent.fleet.preset)}}
+{{- fail (printf "specified preset \"%s\" under fleet is not found" $.Values.agent.fleet.preset) -}}
+{{- end -}}
```

Comment:
Some of the nested bit will remain as we have to validate the fields you mention only when the fleet mode is enabled. I am not a big fun of `required` in this particular case as it will indeed fail if the specified variable is empty but if it is not it will return it and I will have to do an empty assignment ($_) not to have it printed. However, I will switch to it and hopefully the one liner will look better